ScopedStorage/app/src/main/java/com/samples/storage/mediastore/AddMediaViewModel.kt

```diff
@@ -75,22 +85,49 @@ class AddMediaViewModel(application: Application) : AndroidViewModel(application
     }
 
     // We create a URI where the camera will store the video
-    fun createVideoUri(): Uri? {
+    fun createVideoUri(source: Source): Uri? {
         val videoCollection = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
             MediaStore.Video.Media.getContentUri(MediaStore.VOLUME_EXTERNAL_PRIMARY)
         } else {
             MediaStore.Video.Media.EXTERNAL_CONTENT_URI
         }
 
         val newVideo = ContentValues().apply {
-            put(MediaStore.Video.Media.DISPLAY_NAME, generateFilename(Source.CAMERA, "mp4"))
+            put(MediaStore.Video.Media.DISPLAY_NAME, generateFilename(source, "mp4"))
         }
 
         val uri = context.contentResolver.insert(videoCollection, newVideo)
         temporaryMediaUri = uri
 
         return uri
     }
+
+    fun saveRandomImageFromInternet() {
```

Comment:
Please consider disabling the download button while the download is in progress or cancel an ongoing download if the button is pressed a second time.